lib/features/history/models/income.dart

```diff
@@ -10,15 +10,15 @@ class Allowance extends HistoryItem {
   @override
   List<Object?> get props => [amount, date, type];
 
-  Allowance.empty()
+  Income.empty()
       : this(
           amount: 0,
           date: DateTime.now(),
           type: HistoryTypes.donation,
         );
 
-  factory Allowance.fromMap(Map<String, dynamic> map) {
-    return Allowance(
+  factory Income.fromMap(Map<String, dynamic> map) {
+    return Income(
         amount: double.tryParse(map['amount'].toString()) ?? 0,
         date: DateTime.tryParse(map['donationDate']) ?? DateTime.now(),
         type: HistoryTypes.values.firstWhere(
```

Comment:
Related to the other comment. 

Then I would make the orElse of the firstWhere in type return HistoryTypes.unknown/default.